.github/workflows/ci.yml

```diff
@@ -35,15 +35,40 @@ jobs:
       - run: npm run test:lint
       - run: npm run test:unit
 
+  dry-run:
+    # Don't run on pull requests from forks - they don't have access to secrets
+    if: >-
+      github.event_name != 'pull_request' ||
+      github.event.pull_request.head.repo.full_name == github.repository
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_DB: postgres
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_USER: postgres
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+    env:
+      DATABASE_URL: postgres://postgres:postgres@localhost:5432/postgres
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - run: npm ci
+      - run: npm run migrate
       - name: Evaluate the last Spark round in dry-run mode
-        # Don't run on forks, which don't have access to secrets
-        if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository
```

Comment:
I moved this condition to the top of this block so that the entire `dry-run` block is skipped when the condition is false.